src/ir/proto/sql/sql_ir.proto

```diff
@@ -0,0 +1,108 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+// This is a minimal proto targeted at the MVP integration with a Google SQL
+// verifier. It's just to push a limited set of constructs across the
+// language barrier.
+syntax = "proto3";
+
+package raksha.ir.proto.sql;
+
+// This value corresponds to some source table for the query being analyzed.
+message SourceTableColumn {
+  // The fully-qualified path to the column, including table name and (if it
+  // exists) schema name.
+  string column_path = 1;
+}
+
+// A literal. The SQL verifier currently just passes on the string form of the
+// literal as seen by the parser. For the MVP, we just continue passing this
+// along.
+message Literal {
+  string literal_str = 1;
+}
+
+// This `MergeOperation` is an all-purpose combination of inputs into a single
+// output.
+message MergeOperation {
+  // Direct dataflow dependencies. Relevant for confidentiality and integrity
+  // tags.
+  repeated Expression inputs = 2;
+  // Control dependencies. Confidentiality tags can flow from these inputs to
+  // the outputs. They are irrelevant to integrity tags.
+  repeated Expression control_inputs = 3;
+}
+
+// The kind of a tag.
+enum TagKind {
+  UNKNOWN = 0;
+  INTEGRITY = 1;
+  CONFIDENTIALITY = 2;
+}
+
+// The action that is taken upon a tag by a policy rule.
+enum TagAlteration {
+  NONE = 0;
+  ADD = 1;
+  REMOVE = 2;
+}
+
+// A transform upon the state of a program as dictated by a policy rule.
+//
+// A `PolicyTransform` is structured like a cast upon some `transformed_node`
+// `Expression`. The preconditions of this transform, however, are presented as
+// IDs rather than nesting the nodes directly underneath them. This is
+// necessary because these inputs may not be direct children of the
+// transformed node. Because protos only allow each structure to have a
+// single nesting parent, we need a different way of referring to them for
+// these additional references.
+message PolicyTransform {
+  // The `Expression` that is having its tag state changed by the
+  // `PolicyTransform`.
+  Expression transformed_node = 1;
```

Comment:
Is there always one expression? You might want to change this to

```
repeated Expression inputs = 1;
``` 
and use the  index of inputs as the key in `required_integrity_tags`?

Essentially, collect all the required context in this message as `inputs`.

Reply:
There is always one root expression for these transformations. The message currently known as `PolicyTransform` acts as a cast upon the `transformed_node`. It may consider whether `IntegrityTag`s are present upon other `Expression`s, but because they are structurally nested within other `Expression`s (either the `transformed_node` or one of its transitive children) we cannot embed them here as well.